### ISSUE TITLE
Enable metadata support for peers not in mesh

### ISSUE THREAD

**Issue by douglas-reid** [istio/istio#24302]
Currently, v2 telemetry only supports metadata exchange between in-meshed workloads. Need to add support for metadata lookup for non-mesh workloads.

[ X ] Policies and Telemetry

**Comment by douglas-reid** [comment]
Related issue: #20538

**Comment by douglas-reid** [comment]
Not that I'm aware of. There have been multiple discussions in various forums, but I don't know that there has ever been a formal RFC.

**Comment by bianpengyuan** [comment]
Push this to 1.9. We have a short to medium term solution to add metadata into client endpoint resource, which addresses part of this issue.

**Comment by douglas-reid** [comment]
@bianpengyuan can you comment on status of this issue re: release-1.9?

**Comment by bianpengyuan** [comment]
At 1.9, we've finished telemetry metadata for out of mesh server.

We discussed solution for client telemetry metadata at p&t wg meeting, which will have to involve a central component (either istiod or a separated deployment) for metadata query. Considering that customer issues are mostly around missing metadata labels at client side metrics, I will set this feature to nebulous future. If customer demands telemetry metadata for client out-of-mesh, we can evaluate this issue again.
